pandas/core/arrays/masked.py

```diff
@@ -181,7 +182,9 @@ def __getitem__(self, item: SequenceIndexer) -> Self:
     def __getitem__(self, item: PositionalIndexer) -> Self | Any:
         item = check_array_indexer(self, item)
 
-        newmask = self._mask[item]
+        # TODO: need to change this to special case multiple
+        # indexers versus just scalar
+        newmask = self._mask.to_numpy()[item]
```

Comment:
The big blocker to this (as far as current test coverage is concerned) is the case when `item` is a slice. I wasn't really sure how to best work with that in `__getitem__` - do we already have an established pattern for handling this in lib?